lib/Parse/ParseExpr.cpp

```diff
@@ -2689,6 +2689,15 @@ Parser::parseExprObjectLiteral(ObjectLiteralExpr::LiteralKind LitKind,
   if (status.isError())
     return makeParserError();
 
+  // Arguments must be literals and must not be interpolated string literals.
+  for (auto arg : args) {
+    if (!isa<LiteralExpr>(arg) || isa<InterpolatedStringLiteralExpr>(arg)) {
+      diagnose(arg->getLoc(),
+               diag::expected_uninterpolated_literal_arg_in_object_literal);
+      return makeParserError();
```

Comment:
Can we make this error being conditional on Swift 4 mode, leaving this a warning in Swift 3 compatibility mode?  That would give people a heads up that this behavior is going away without it being a hard break.